modules/host/network.go

```diff
@@ -49,11 +49,11 @@ func (h *Host) initNetworking(address string) (err error) {
 		return err
 	}
 	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return err
+	}
 	go func() {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
```

Comment:
Goroutines are not guaranteed to be run in the same order that they are created.

This still wouldn't cause a timeout during shutdown though. Either this thread is has called `Add` which means the `Close` thread is going to block, or this thread has not yet called `Add` which means the `Close` thread is not going to block. What you've done is guarantee that `Add` has been called, which can only increase the length of time that it takes for shutdown to occur.

The host tests are timing out during the short tests or the long tests? If there is a deadlock in the host, it's not here.

In the parent PR, can you provide some of the dumps associated with the timeouts? It might just be the short vs. long test thing again.